pkg/printers/name.go

```diff
@@ -62,26 +63,34 @@ func (p *NamePrinter) PrintObj(obj runtime.Object, w io.Writer) error {
 		}
 	}
 
-	kind := obj.GetObjectKind().GroupVersionKind()
-	if len(kind.Kind) == 0 {
-		if gvks, _, err := p.Typer.ObjectKinds(obj); err == nil {
-			for _, gvk := range gvks {
-				if mappings, err := p.Mapper.RESTMappings(gvk.GroupKind(), gvk.Version); err == nil && len(mappings) > 0 {
-					fmt.Fprintf(w, "%s/%s\n", mappings[0].Resource, name)
-				}
+	groupVersionKind := obj.GetObjectKind().GroupVersionKind()
+	if len(groupVersionKind.Kind) > 0 {
+		if mappings, err := p.Mapper.RESTMappings(groupVersionKind.GroupKind(), groupVersionKind.Version); err == nil && len(mappings) > 0 {
+			fmt.Fprintf(w, "%s/%s\n", mappings[0].Resource, name)
+			return nil
+		}
+	}
+
+	if gvks, _, err := p.Typer.ObjectKinds(obj); err == nil {
+		for _, gvk := range gvks {
+			if mappings, err := p.Mapper.RESTMappings(gvk.GroupKind(), gvk.Version); err == nil && len(mappings) > 0 {
+				fmt.Fprintf(w, "%s/%s\n", mappings[0].Resource, name)
+				return nil
 			}
-		} else {
-			fmt.Fprintf(w, "<unknown>/%s\n", name)
 		}
+	}
 
-	} else {
-		if mappings, err := p.Mapper.RESTMappings(kind.GroupKind(), kind.Version); err == nil && len(mappings) > 0 {
-			fmt.Fprintf(w, "%s/%s\n", mappings[0].Resource, name)
-		} else {
-			fmt.Fprintf(w, "<unknown>/%s\n", name)
+	if unstructured, ok := obj.(runtime.Unstructured); ok {
+		unstructuredContent := unstructured.UnstructuredContent()
+		if unstructuredKind, ok := unstructuredContent["kind"]; ok {
```

Comment:
How about gathering the groupversionkind and use those to lookup the restmapping?  The restmapping should exist since we were able to lookup the resource URL right?